packages/build/src/plugins/spawn.ts

```diff
@@ -35,25 +43,72 @@ const tStartPlugins = async function ({ pluginsOptions, buildDir, childEnv, logs
   logIncompatiblePlugins(logs, pluginsOptions)
 
   const childProcesses = await Promise.all(
-    pluginsOptions.map(({ pluginDir, nodePath, pluginPackageJson }) =>
-      startPlugin({ pluginDir, nodePath, buildDir, childEnv, systemLogFile, pluginPackageJson }),
+    pluginsOptions.map(({ pluginDir, nodePath, nodeVersion, pluginPackageJson }) =>
+      startPlugin({ pluginDir, nodePath, nodeVersion, buildDir, childEnv, systemLogFile, pluginPackageJson }),
     ),
   )
   return { childProcesses }
 }
 
 export const startPlugins = measureDuration(tStartPlugins, 'start_plugins')
 
-const startPlugin = async function ({ pluginDir, nodePath, buildDir, childEnv, systemLogFile, pluginPackageJson }) {
-  const childProcess = execaNode(CHILD_MAIN_FILE, [], {
+const startPlugin = async function ({
+  pluginDir,
+  nodeVersion,
+  nodePath,
+  buildDir,
+  childEnv,
+  systemLogFile,
+  pluginPackageJson,
+}: {
+  nodeVersion: string
+  nodePath: string
+  pluginDir: string
+  /** The process cwd that is used to spawn the child process */
+  buildDir: string
+  childEnv: Record<string, string>
+  pluginPackageJson: Record<string, string>
+  systemLogFile: number
+}) {
+  const ctx = trace.getActiveSpan()?.spanContext()
+
+  // the baggage will be passed to the child process when sending the run event
+  const args = [
+    ...process.argv.filter((arg) => arg.startsWith('--tracing')),
+    `--tracing.traceId=${ctx?.traceId}`,
+    `--tracing.parentSpanId=${ctx?.spanId}`,
+    `--tracing.traceFlags=${ctx?.traceFlags}`,
+    `--tracing.enabled=${!!isTrustedPlugin(pluginPackageJson?.name)}`,
+  ]
+
+  const nodeOptions: string[] = []
+
+  // the sdk setup is a optional dependency that might not exist
+  // only use it if it exists
+  try {
+    // the --import preloading is only available in node 18.18.0 and above
+    // plugins that run on a lower node version will not be able to be instrumented with opentelemetry
+    if (gte(nodeVersion, '18.18.0')) {
+      const entry = require.resolve('@netlify/opentelemetry-sdk-setup/bin.js')
+      // on windows only file:// urls are allowed
+      nodeOptions.push('--import', pathToFileURL(entry).toString())
+    }
+  } catch {
+    // noop
+  }
+
+  const childProcess = execaNode(CHILD_MAIN_FILE, args, {
     cwd: buildDir,
     preferLocal: true,
     localDir: pluginDir,
     nodePath,
-    // make sure we don't pass build's node cli properties for now (e.g. --import)
-    nodeOptions: [],
+    nodeOptions,
     execPath: nodePath,
-    env: childEnv,
+    env: {
+      ...childEnv,
+      OTEL_SERVICE_NAME: pluginPackageJson?.name,
+      OTEL_SERVICE_VERSION: pluginPackageJson?.version,
```

Comment:
Do we also need this together with?
- https://github.com/netlify/build/blob/6a696a928dcf94ae8ce63e7b379ce1efb8386b62/packages/opentelemetry-sdk-setup/src/sdk-setup.ts#L38-L45

I'm under the impression that the env vars will override our programmatic definition in the SDK properties.

Reply:
The semantic resources are the way to go. I just kept both for now. (should not matter or harm us)